website/photos/models.py

```diff
@@ -27,7 +27,8 @@
 
 def photo_uploadto(instance, filename):
     ext = os.path.splitext(filename)[1]
-    return f"photos/{instance.album.dirname}/{token_hex(8)}{ext}"
+    position = str(instance).split("-")[0]
```

Comment:
This is weird:

```
    def __str__(self):
        """Return the filename of a Photo object."""
        return os.path.basename(self.file.name)
```

That wouldn't have a - at all, and seems like a circular dependency to check the file name from uploadto

Reply:
Ah, I see, you're adding it in the filename in _try_save_photo Can you just get the index in this function directly instead of expecting the filename to have a specific format?